### PR TITLE
fix: check user/group in container start

### PR DESCRIPTION
**Pull request by Ace-Tang**
user parameter passed by pouch run --user is not validate in container
start process, since we just try to find file in BaseFS, but BaseFS is
created in containerd, so we pass the validate. Fix this by find file in
image container use.
Add function Get which has GetUser and GetAdditionalGids function as a
interface in pkg/user package.

Signed-off-by: Ace-Tang <aceapril@126.com>

### CODE REVIEW

File: daemon/mgr/container_types.go [hunk 1]
```diff
@@ -352,6 +355,46 @@ func (c *Container) UnsetMergedDir() {
...
+
+	c.Snapshotter = &types.SnapshotterData{
+		Name: "overlayfs",
+		Data: data,
+	}
+}
+
+// GetSpecificBasePath accepts a given path, look for whether the path is exist
+// within container, if has, returns container base path like BaseFS, if not, return empty string
+func (c *Container) GetSpecificBasePath(path string) string {
```
Comment (HusterWan) [inline daemon/mgr/container_types.go]: feel weird ! can we judge whether the baseFS is mounted?

Comment (HusterWan) [reply daemon/mgr/container_types.go]: i still have another question: can we ensure passwd file exists in UpperDir? 

Or can we mount the mergedDir when we found it not mounted, then we can ensure the passwd file exists in MergedDir.

it's just my advice, you can think about it again, thanks a lot @Ace-Tang

Comment (Ace-Tang) [reply daemon/mgr/container_types.go]: when we run a container, merged dir is not created yet until pouch pass create options to containerd, then containerd created it and mount. Add this function cause we can only find file in image.

Comment (Ace-Tang) [reply daemon/mgr/container_types.go]: I also think this pr can fix if merged dir is un-mounted by error.

File: daemon/mgr/container_types.go [hunk 2]
```diff
@@ -5,13 +5,16 @@ import (
...
+	"path/filepath"
+	"strings"
 	"sync"
 	"time"
 
 	"github.com/alibaba/pouch/apis/types"
 	"github.com/alibaba/pouch/cri/stream/remotecommand"
 	"github.com/alibaba/pouch/pkg/meta"
 	"github.com/alibaba/pouch/pkg/utils"
+	"github.com/containerd/containerd/mount"
```
Comment (HusterWan) [inline daemon/mgr/container_types.go]: please put this line to third-party package

File: daemon/mgr/container_types.go [hunk 3]
```diff
@@ -352,6 +355,46 @@ func (c *Container) UnsetMergedDir() {
...
+
+// GetSpecificBasePath accepts a given path, look for whether the path is exist
+// within container, if has, returns container base path like BaseFS, if not, return empty string
+func (c *Container) GetSpecificBasePath(path string) string {
+	// first try container BaseFS, it is a general view,
+	if utils.IsFileExist(filepath.Join(c.BaseFS, path)) {
+		return c.BaseFS
+	}
+
+	// then try lower and upper directory, since overlay filesystem support only.
```
Comment (HusterWan) [inline daemon/mgr/container_types.go]: a good solution to fix mergeDir not mounted 👋

**Comment by Ace-Tang** [comment]
@HusterWan ,Updated. Beside, I have test a occasion that umount container rootfs to see if exec will successful.
```
#./pouch ps
Name     ID       Status        Created      Image                                          Runtime
f7fdec   f7fdec   Up 17 hours   3 days ago   registry.hub.docker.com/library/busybox:1.25   runc

#./pouch exec f7fdec echo 1
1

#mount -l | grep overlay
/dev/sda5 on /home/docker/overlay type ext4 (rw,relatime,grpquota,data=ordered) [/home]
/dev/sda5 on /home/docker/overlay2 type ext4 (rw,relatime,grpquota,data=ordered) [/home]
overlay on /home/docker/overlay/f894c1a86b8cc750e44ec8bd42242c06def8f3ff33b36c3ba323374dcfe58a8d/merged type overlay (rw,relatime,lowerdir=/home/docker/overlay/8aa06fac0dc3119bc3da5fd19ce08270346e06757cbe964152fc7e179bce5fe6/root,upperdir=/home/docker/overlay/f894c1a86b8cc750e44ec8bd42242c06def8f3ff33b36c3ba323374dcfe58a8d/upper,workdir=/home/docker/overlay/f894c1a86b8cc750e44ec8bd42242c06def8f3ff33b36c3ba323374dcfe58a8d/work)
overlay on /var/lib/pouch/containerd/state/io.containerd.runtime.v1.linux/default/f7fdec4eff4f4206de5424f0c4f5b8498ba63b09e53ebc3125c03107cdf35490/rootfs type overlay (rw,relatime,lowerdir=/var/lib/pouch/containerd/root/io.containerd.snapshotter.v1.overlayfs/snapshots/1/fs,upperdir=/var/lib/pouch/containerd/root/io.containerd.snapshotter.v1.overlayfs/snapshots/513/fs,workdir=/var/lib/pouch/containerd/root/io.containerd.snapshotter.v1.overlayfs/snapshots/513/work)

#umount /var/lib/pouch/containerd/state/io.containerd.runtime.v1.linux/default/f7fdec4eff4f4206de5424f0c4f5b8498ba63b09e53ebc3125c03107cdf35490/rootfs

#mount -l | grep overlay
/dev/sda5 on /home/docker/overlay type ext4 (rw,relatime,grpquota,data=ordered) [/home]
/dev/sda5 on /home/docker/overlay2 type ext4 (rw,relatime,grpquota,data=ordered) [/home]
overlay on /home/docker/overlay/f894c1a86b8cc750e44ec8bd42242c06def8f3ff33b36c3ba323374dcfe58a8d/merged type overlay (rw,relatime,lowerdir=/home/docker/overlay/8aa06fac0dc3119bc3da5fd19ce08270346e06757cbe964152fc7e179bce5fe6/root,upperdir=/home/docker/overlay/f894c1a86b8cc750e44ec8bd42242c06def8f3ff33b36c3ba323374dcfe58a8d/upper,workdir=/home/docker/overlay/f894c1a86b8cc750e44ec8bd42242c06def8f3ff33b36c3ba323374dcfe58a8d/work)

#./pouch exec f7fdec echo 1
1
```

**Comment by HusterWan** [comment]
lgtm

**Pull request merged**